com.unity.shadergraph/Editor/ShaderFoundry/Samples/SimpleSampleBuilder.cs

```diff
@@ -16,11 +16,12 @@ internal static void Build(ShaderContainer container, string shaderName, BuildCa
 
         internal static void Build(ShaderContainer container, Target target, string shaderName, BuildCallback buildCallback, ShaderBuilder shaderBuilder)
         {
+            // ideally the settings for this provider would be provided through the same interface as workflow settings
+            // but for now it is directly populated by this special constructor
             ITemplateProvider provider = new LegacyTemplateProvider(target, new ShaderGraph.AssetCollection());
 
-            var shaderInstBuilder = new ShaderInstance.Builder(container, shaderName);
-            
-            foreach(var template in provider.GetTemplates(container))
+            var shaderInstBuilder = new ShaderInstance.Builder(container, shaderName);            
+            foreach (var template in provider.GetTemplates(container))
```

Comment:
Each template can go to a separate shader instance right? What's controlling this?

Reply:
Fixed so SimpleShaderBuilder can enumerate and invoke additional shader ids